### ISSUE TITLE
Array return value is broken when returning null

### ISSUE THREAD

**Issue by alexvas** [typetools/checker-framework#6676]
### Commands 
```bash
java -version -Aversion -AprintGitProperties -verbose -AprintVerboseGenerics
java -jar "$CHECKER_FRAMEWORK/checker/dist/checker.jar" -processor nullness Foo.java
```
 ### Inputs
A single input named `Foo.java`:
```java
import org.checkerframework.checker.nullness.qual.Nullable;

interface Foo {
}

interface Bar {

	default @Nullable Foo[] bazz() {
		return null;
	}
}
```
 ### Outputs
```
openjdk version "17.0.11" 2024-04-16
OpenJDK Runtime Environment (build 17.0.11+9-Ubuntu-122.04.1)
OpenJDK 64-Bit Server VM (build 17.0.11+9-Ubuntu-122.04.1, mixed mode, sharing)
Foo.java:9: error: [return] incompatible types in return.
                return null;
                       ^
  type of expression: null (NullType)
  method return type: @Initialized @Nullable Foo @Initialized @NonNull []
1 error
```
 ### Expectation
The case is valid. No error should be issued.

**Comment by msridhar** [comment]
I think you need to write `Foo @Nullable []` since it's a type use annotation

**Comment by alexvas** [comment]
@msridhar, Thank you. My bad.
